src/display/editor/annotation_editor_layer.js

```diff
@@ -809,8 +812,18 @@ class AnnotationEditorLayer {
   }
 
   createAnnotationNode() {
-    const { event, data } = this.#highlightSelectionParams;
-    this.#createAndAddNewEditor(event, false, data);
+    const editorType = this.#currentEditorType;
+    if (editorType) {
+      const { event, data } = this.#highlightSelectionParams;
+      this.#createAndAddNewEditor(event, false, data);
+      this.#highlightSelectionParams = null;
+    } else {
+      const editor = new HighlightEditor.prototype.constructor({
+        ...this.#linkNodeParams,
+      });
+      this.add(editor);
+      this.#linkNodeParams = null;
```

Comment:
Same here

Reply:
But this one is not added to this.removeTempHighlight(). Do I need to add this also?

Reply:
No bro, you are right, leave it as is